project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/services/MinimizationPluginSettings.kt

```diff
@@ -0,0 +1,18 @@
+package org.plan.research.minimization.plugin.services
+
+import org.plan.research.minimization.plugin.settings.MinimizationPluginState
+import org.plan.research.minimization.plugin.settings.MinimizationPluginStateObservable
+
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.components.SimplePersistentStateComponent
+import com.intellij.openapi.components.State
+
+@Service(Service.Level.PROJECT)
+@State(name = "MinimizationPluginSettings")
+class MinimizationPluginSettings : SimplePersistentStateComponent<MinimizationPluginState>(MinimizationPluginState()) {
+    var stateObservable: MinimizationPluginStateObservable = MinimizationPluginStateObservable(state)
```

Comment:
Here could be a problem: somewhere in IDEA could happen something like `project.service<MinimizationPluginSettings>().state = <some new state>`, but `stateObservable` will use the old version of `state`.

Reply:
I'm not sure about this case, but let's make solution independent of such cases